src/core/sensor/qgsiodevicesensor.cpp

```diff
@@ -186,7 +186,7 @@ QgsUdpSocketSensor::QgsUdpSocketSensor( QObject *parent )
 #endif
 
   connect( mUdpSocket.get(), &QAbstractSocket::stateChanged, this, &QgsUdpSocketSensor::socketStateChanged );
-  connect( mUdpSocket.get(), &QUdpSocket::readyRead, this, [this]()
+  connect( mUdpSocket.get(), &QUdpSocket::readyRead, this, [ = ]()
```

Comment:
actually the original form is correct -- implicate capture of this is deprecated in c++